test/ListProxyTest.js

```diff
@@ -0,0 +1,266 @@
+            return listInstance.add(1);
+        }).then(function () {
+            return listInstance.isEmpty().then(function (empty) {
+                expect(empty).to.be.false;
+            })
+        })
+    });
+
+    it("removes an entry", function () {
+        return listInstance.addAll([1, 2, 3]).then(function () {
+            return listInstance.remove(1)
+        }).then(function () {
+            return listInstance.toArray().then(function (all) {
+                expect(all).to.deep.equal([2, 3]);
+            });
+        });
+    });
+
+    it("removes an entry by index", function () {
+        return listInstance.addAll([1, 2, 3]).then(function () {
+            return listInstance.removeAt(1)
+        }).then(function () {
+            return listInstance.toArray().then(function (all) {
+                expect(all).to.deep.equal([1, 3]);
+            });
+        });
+    });
+
+    it("removes multiple entries", function () {
+        return listInstance.addAll([1, 2, 3, 4]).then(function () {
+            return listInstance.removeAll([1, 2]);
+        }).then(function () {
+            return listInstance.toArray().then(function (all) {
+                expect(all).to.deep.equal([3, 4]);
+            });
+        });
+    });
+
+    it("retains multiple entries", function () {
+        return listInstance.addAll([1, 2, 3, 4]).then(function () {
+            return listInstance.retainAll([1, 2]);
+        }).then(function () {
+            return listInstance.toArray().then(function (all) {
+                expect(all).to.deep.equal([1, 2]);
+            });
+        });
+    });
+
+    it("finds index of the element", function () {
+        return listInstance.addAll([1, 2, 4, 4]).then(function () {
+            return listInstance.indexOf(4).then(function (index) {
+                expect(index).to.equal(2);
+            });
+        });
+    });
+
+    it("finds last index of the element", function () {
+        return listInstance.addAll([1, 2, 4, 4]).then(function () {
+            return listInstance.lastIndexOf(4).then(function (index) {
+                expect(index).to.equal(3);
+            });
+        });
+    });
+
+    it("returns a sub list", function () {
+        return listInstance.addAll([1, 2, 3, 4, 5, 6]).then(function () {
+            return listInstance.subList(1, 5).then(function (subList) {
+                expect(subList).to.deep.equal([2, 3, 4, 5]);
+            });
+        });
+    });
+
+    it("listens for added entry", function (done) {
+        this.timeout(5000);
+        var listener = {
+            "itemAdded" : function (item) {
+                if (item == 1) {
+                    done()
+                } else {
+                    done(new Error("Expected 1, got " + item))
+                }
+            }
+        };
+        return listInstance.addItemListener(listener, true).then(function () {
+            listInstance.add(1);
+        })
+    });
+
```

Comment:
Can you add another test where `includeValue` for ItemListener is `false`. Then branch coverage for this PR will be complete.